native/cocos/core/assets/Texture2D.cpp

```diff
@@ -40,6 +40,33 @@ void Texture2D::syncMipmapsForJS(const ccstd::vector<IntrusivePtr<ImageAsset>> &
 }
 
 void Texture2D::setMipmaps(const ccstd::vector<IntrusivePtr<ImageAsset>> &value) {
+    if (!value.empty()) {
+        _compressedImageAsset.clear();
+        const auto mipmapLevelData = value[0]->getMipmapLevelDataSize();
+        _compressedImageAsset.resize(mipmapLevelData.size());
+        auto *data = const_cast<unsigned char *>(value[0]->getData());
+
+        uint32_t byteOffset = 0;
+        for (uint32_t i = 0; i < mipmapLevelData.size(); ++i) {
+            auto *dstData = static_cast<unsigned char *>(malloc(mipmapLevelData[i] * sizeof(unsigned char)));
+            memcpy(dstData, data + byteOffset, mipmapLevelData[i]);
+
+            _compressedImageAsset[i] = new ImageAsset();
+            _compressedImageAsset[i]->setData(dstData);
```

Comment:
Need to add `ImageAsset::setNeedFreeData(bool v)` method and invoking `_compressedImageAsset[i]->setNeedFreeData(true)` after setData to avoid memory leak here.